dbms/src/Processors/Formats/Impl/CSVRowInputFormat.cpp

```diff
@@ -433,4 +433,67 @@ void registerInputFormatProcessorCSV(FormatFactory & factory)
     }
 }
 
+void registerFileSegmentationEngineCSV(FormatFactory & factory)
+{
+    factory.registerFileSegmentationEngine("CSV", [](
```

Comment:
Could you make these plain functions instead of lambdas? It's better for call stacks.